channel/allocation.go

```diff
@@ -95,8 +95,14 @@ type (
 	// Asset identifies an asset. E.g., it may be the address of the multi-sig
 	// where all participants' assets are deposited.
 	// The same Asset should be shareable by multiple Allocation instances.
-	// Decoding happens with AppBackend.DecodeAsset.
-	Asset = encoding.BinaryMarshaler
+	Asset interface {
+		// BinaryMarshaler marshals the blockchain specific address to
+		// binary format (a byte array).
+		encoding.BinaryMarshaler
+		// BinaryUnmarshaler unmarshals the blockchain specific address
+		// from binary format (a byte array).
+		encoding.BinaryUnmarshaler
```

Comment:
Would it make sense to create a combined interface like `BinaryMarshalerUnmarshaler` or something like that?

Reply:
Though about this and even tried.

But, the tricky part was: "where to put this combined interface" ?

Semantically it fits into the wire package (because it represents that a data will be encoded as binary data when sending over the wire). 

But, putting this in wire causes import cycles (because wire uses wallet.Address and address implements wire.Binary)

Cannot put it in io package, because the binary interface is not specific for perun encoder (implemented in pkg/io) package. But, it is applicable for other encoders that will be added in future.

The only other option, I could think of was to define the interface twice (in wire and channel packages)

What do you think, we should be doing ?

Reply:
I see, BinaryMarshaler and BinaryUnmarshaler come from a go std package, but that one doesn't provide BinaryMarshalerUnmarshaler. In this case I think it is fine. We can leave it like it is.